WordPress/wordpress-develop
# Fix minor bug reintroduced in refactoring and introduction of `wp_get_loading_optimization_attributes()`

Trac ticket: https://core.trac.wordpress.org/ticket/58235

See https://core.trac.wordpress.org/ticket/58235#comment:33:
* This PR fixes a regression related to what was originally fixed by https://core.trac.wordpress.org/changeset/55825.
* It also adjusts related tests which were expecting incorrect behavior: Those special contexts should never receive any modifications, not only `loading="lazy"`, but also `fetchpriority="high"`.

### Testing instructions
* Using a classic theme, create a new post with 3 "large" (or larger than the "large" image size) images.
* Add the following snippet to your site (make sure to update the one instance per the comment):
```php
add_filter(
	'the_content',
	function( $content ) {
		// Replace this with some actual image attachment available on your test site.
		$attachment_id = 0;
		$content      .= wp_get_attachment_image( $attachment_id, 'large' );
		return $content;
	},
	9
);
```
* The above is just one way to trigger the bug. You could alternatively put a featured image block to the bottom of the post content.
* Visit the single post URL.
* Without this PR, you should see that the last image (added via the code snippet) has `fetchpriority="high"` and is not lazy-loaded, which also means that the actual first image does not have `fetchpriority="high"`. This is incorrect and happens because the last image is separately parsed before the rest of the content has been parsed.
* With this PR applied, it should be fixed, so that the first in-content image has `fetchpriority="high"` instead of the last one. This should also lead to the last image being lazy-loaded.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
